src/hotspot/share/nmt/virtualMemoryTracker.hpp

```diff
@@ -372,40 +366,60 @@ class VirtualMemoryWalker : public StackObj {
    virtual bool do_allocation_site(const ReservedMemoryRegion* rgn) { return false; }
 };
 
-// Main class called from MemTracker to track virtual memory allocations, commits and releases.
-class VirtualMemoryTracker : AllStatic {
-  friend class VirtualMemoryTrackerTest;
-  friend class CommittedVirtualMemoryTest;
+class RegionsTree;
+class VirtualMemoryTracker {
+ private:
+  RegionsTree *_tree;
```

Comment:
`class RegionsTree;` shouldn't be needed if you fix the circular include from above. There is no need to have the `private:` specifier. The `_tree` doesn't need to be a pointer after the forward declaration is removed, which in turn simplifies the initialization code.

Reply:
new `regionsTree.inline.hpp` is introduced to resolve the dependencies.